docs/developer-docs/smart-contracts/maintain/trapping.mdx

```diff
@@ -29,42 +35,31 @@ If a canister traps during a message execution, then two things happen:
 Using pre_ and post_upgrade hooks is discouraged. It is error-prone and can render a canister unusable. Per best practices, using these methods should be avoided if possible.
 :::
 
-Special care should be taken when writing a `pre_upgrade` hook for your canister which will run during an upgrade. If the [pre_upgrade](/docs/current/references/ic-interface-spec#system-api-upgrades) hook traps,
-then the canister upgrade would fail. This is extremely important as it can mean your canister can be permanently bricked if such a trap occurs
-in the `pre_upgrade` hook as this will always run against the current canister's Wasm module.
+Special care should be taken when writing a `pre_upgrade` hook for your canister which will run during an upgrade against the current canister's Wasm module. If the [pre_upgrade](/docs/current/references/ic-interface-spec#system-api-upgrades) hook traps, then the canister upgrade will fail. This is extremely important as it can mean your canister can be permanently unrecoverable.
```

Comment:
```suggestion
Special care should be taken when writing a `pre_upgrade` hook for your canister which will run during an upgrade against the current canister's Wasm module. If the [`pre_upgrade`](/docs/current/references/ic-interface-spec#system-api-upgrades) hook traps, then the canister upgrade will fail. This is extremely important as it can mean your canister can be permanently unrecoverable.
```